src/app/editor/shared/services/schema-validation.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class SchemaValidationService {
+  
+  private static formatRegexps = {
+    date: /^\d\d\d\d-[0-1]\d-[0-3]\d$/,
+    email: /^[a-z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?)*$/i,
+    isbn: /^ISBN\s(?=[-0-9xX ]{13}$)(?:[0-9]+[- ]){3}[0-9]*[xX0-9]$/,
```

Comment:
Ah right, I was thinking about the `pattern` key I had seen https://github.com/inspirehep/record-editor/blob/master/src/assets/hep.json#L1059 I guess the `pattern` will be taken care of by the schema validation library.

Are using any third party schema validation at the moment?